sentry-spring/src/main/java/io/sentry/spring/SentryFilterCachingRequest.java

```diff
@@ -0,0 +1,21 @@
+package io.sentry.spring;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class SentryFilterCachingRequest extends OncePerRequestFilter {
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        ContentCachingRequestWrapper cacheRequest = new ContentCachingRequestWrapper(request);
+        filterChain.doFilter(cacheRequest, response);
+    }
```

Comment:
> While this is a very nice solution, I'm a little bit unsure about the effects of requiring the content caching wrapper of the request unconditionally. Is there a situation where this might blow up memory consumption?

@metlos Thanks so much for the review! At the expense of memory, an explosion does occur if the request body is binary and large, as well as if the request was already in the cache. I tried to solve this by checking that the request came in was an instance of the ContentCachingRequestWrapper class, and also doing a check on the size of the request body, however, I can’t decide what value is optimal for caching, the Sentry documentation says something about 10 kilobytes per message, however can you suggest another optimal value

Example code 
``` 
 private final long MAX_BODY_SIZE_CACHING = 31457280; //~30 mb
    @Override
    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
            throws ServletException, IOException {
        if (request instanceof ContentCachingRequestWrapper || request.getContentLengthLong() >= MAX_BODY_SIZE_CACHING){
            filterChain.doFilter(request, response);
            return;
        }
        ContentCachingRequestWrapper wrappedRequest = new ContentCachingRequestWrapper(request);
```

Reply:
Most likely, will have to specify the priority of the filter execution, namely, assign the lowest priority so that all user filters are executed first

Reply:
I would also think about introducing some kind of blacklist for some Rest paths at the user's choice, we will offer an empty blacklist and an interface that can be implemented, the user will write the desired blacklist configuration with path checking and implement it, and redefine the standard configuration in the case of Spring boot or define it yourself in case of vanilla Spring

What do you think about this?

Reply:
> 
> 
> Most likely, will have to specify the priority of the filter execution, namely, assign the lowest priority so that all user filters are executed first

On the other hand, it would be cool to have it working for exceptions raised in user-filters as well. What's wrong with executing this filter first?

The idea with checking content length seems fine, it's better to have it working for majority of cases than none. It would be great to have MB threshold configurable in properties file (along with blacklist you mentioned).

What's the status of it? Are we waiting for review? Anyone?

Reply:
The other implementations of this have "sizes". And also, are opt-in since in many cases request body is read from a stream and never really cached so doing this cache by default could be suboptimal.

Here's how it was done in C#:

https://github.com/getsentry/sentry-dotnet/blob/46bf5548d347ad1e206aefa3ea91cc46ab02004f/src/Sentry/Extensibility/RequestBodyExtractionDispatcher.cs#L43-L69

The idea is that unless a size is specified via `SentryOptions` we're not buffering request bodies.